src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImpl.java

```diff
@@ -71,6 +72,17 @@ public boolean hasOpenPullRequests(String branchName, BitbucketSCMRepository rep
         return pullRequest.stream().filter(pr -> pr.getId() == pullRequestId).findFirst();
     }
 
+    @Override
+    public void refreshStore(String key, String slug, String serverId, List<BitbucketPullRequest> bbsPullRequests) {
+        PullRequestStoreImpl.CacheKey cacheKey =
+                new PullRequestStoreImpl.CacheKey(key, slug, serverId);
+        ConcurrentLinkedQueue<BitbucketPullRequest> pullRequest = pullRequests.get(cacheKey);
```

Comment:
I think we need to re-design our pullrequest store a bit. This method is highly unsafe in a concurrent world. We should do that as a separate PR though, so no need to stress this now.